payroll/views.py

```diff
@@ -541,6 +543,7 @@ def get_context_data(self, **kwargs):
         return context
 
 
+@method_decorator(vary_on_cookie, name='dispatch')
```

Comment:
the search can be one of the most intensive loads on the site, so caching the page per-user sort of erases the benefit, i think... let's talk more about this tuesday. i don't remember having issues when implementing this clientside, but there could be an intermediate solution that doesn't require doing it all over.